assets/js/assessment_v2.js

```diff
@@ -1,185 +1,185 @@
+      }
+      updateQuestions();
+    });
+  });
+};
+
+const populateQuestions = () => {
+  let num = 0;
+  myQuestions.forEach((currentQuestion) => {
+    if (difficultyLevels.indexOf(currentQuestion.difficulty) === -1) {
+      currentQuestion.difficulty = "beginner";
+    }
+    if (!(currentQuestion.difficulty in questions)) {
+      questions[currentQuestion.difficulty] = [];
+    }
+    questions[currentQuestion.difficulty].push(currentQuestion);
+
+    currentQuestion.num = num;
+    num += 1;
+  });
+
+  if (Object.keys(questions).length > 2) {
+    document.getElementById("difficulty-label").style.display = "flex";
+    difficultyLevels.forEach((diff) => {
+      if (!(diff in questions)) {
+        return;
+      }
+      difficulty.push(diff);
+      let checkbox = document.getElementById(diff);
+      checkbox.checked = true;
+      checkbox.parentElement.style.display = "flex";
+    });
+  } else {
+    difficultyLevels.forEach((diff) => {
+      if (!(diff in questions)) {
+        return;
+      }
+      difficulty.push(diff);
+    });
+  }
+};
+
+const checkDifficulties = (classlist) => {
+  if (difficulty.length === Object.keys(questions).length - 1) return true;
+  for (let i in difficulty) {
+    if (classlist.contains(difficulty[i])) return true;
+  }
+  // If beginner is checked list the unlisted question as beginner
+  for (let i in difficultyLevels) {
+    if (classlist.contains(difficultyLevels[i])) return false;
+  }
+  if (difficulty.indexOf("beginner") > -1) {
+    return true;
+  }
+};
+
+function updateQuestions() {
+  const quiz = document.getElementById("quiz");
+  const qquestions = quiz.getElementsByClassName("question");
+  for (let i = 0; i < qquestions.length; i += 1) {
+    if (!checkDifficulties(qquestions[i].classList)) {
+      qquestions[i].style.display = "none";
+      qquestions[i].nextElementSibling.style.display = "none";
+    } else {
+      qquestions[i].style.display = "block";
+      qquestions[i].nextElementSibling.style.display = "flex";
+    }
+  }
+}
+
+function showResults() {
+  // gather answer containers from our quiz
+   const answerContainers = quizContainer.querySelectorAll(".answers");   
+  // keep track of user's answers
+  let numCorrect = 0;
+  let totalNum = 0;
+    
+  // for each question...
+  myQuestions.forEach((currentQuestion) => {
+    // find selected answer
+    if (
+      difficulty.indexOf(currentQuestion.difficulty) === -1 &&
+      difficulty.length !== Object.keys(questions).length - 1
+    )
+      return;
+    let questionNumber = currentQuestion.num;
+    const answerContainer = answerContainers[questionNumber];
+    const selector = `input[name=question${questionNumber}]:checked`;
+    const userAnswer = (answerContainer.querySelector(selector) || {}).value;
+    // Add to total
+    totalNum++;
+
+    // if answer is correct
+    if (userAnswer === currentQuestion.correctAnswer) {
+      // Color the correct answer lightgreen

```

Comment:
The code that was marking all options in black has been removed. What happens when the user submits a quiz once, then goes back, changes some of the options and submits it again.
In this case some of the wrong options would be marked red and others would be marked black.
Please test this scenario before we can move ahead with this PR.

Reply:
In this case some of the wrong options would be marked red and others would be marked black.>>> yes, it does exactly the same. That way the user understands the number of wrong attempts made. If the user reaches the right answer, that will also made green. I have followed up the previous green-red coloring logic.

Reply:
I kind of agree with your argument. Let's run it through Priya once to see what she thinks about it.